source/docs/advanced/theme/server-side-rendering.md

```diff
@@ -0,0 +1,120 @@
+
+## There is no `window` on the server!
+
+Even though it may seem obvious, **you are very likely to come across this problem at some point!** Browser APIs are not available during SSR. Keep it in mind when using them, and provide a graceful fallback.
+
+Sometimes, it could be as simple as a default value or returning earlier from your function without any side effect.
+
+Examples:
+
+```js
+const doSomethingWithIntersectionObserver = () => {
+  if (typeof window === "undefined" || !window.IntersectionObserver) {
+    return;
+  }
+  // …
+}
+```
+
+```js
+const initializePosition = setPosition => {
+  if (typeof navigator === "undefined" || !("geolocation" in navigator)) {
+    setPosition(DEFAULT_LATITUDE, DEFAULT_LONGITUDE);
+  } else {
+    navigator.geolocation.getCurrentPosition(position => {
+      setPosition(position.coords.latitude, position.coords.longitude);
+    });
+  }
+}
+```
+
+## `branchServerClient` at the rescue!
+
+There may be situations were displaying a totally different component during the server rendering might be relevant.
+
+For instance, instead of displaying a map centered on the user geolocation it might be better to render a placeholder (page skeleton) on the server and load the map with additional data during <abbr title="Client Side Rendering">CSR</abbr>. Another use case would be to avoid an unnecessary overhead by not rendering some components during SSR: a social media feed may not make sense on the server.
+
+Front-Commerce provides a `branchServerClient` <abbr title="Higher Order Component">HOC</abbr> that allow you to conditionally render a component or another one on server and client side.
+
+Example:
+
+```js
+import branchServerClient from "web/core/branchServerClient";
+
+const MyClientSideComponent => props => <div>Hi!</div>;
+
+export default branchServerClient(
+  () => () => null, // do not render anything during SSR
+  BaseComponent => BaseComponent // render the base component on the client
+)(MyClientSideComponent);
+```
+
+If you only need to know the rendering context, you can use Front-Commerce's `withIsServer` HOC to have an `isServer` prop injected in your component.
+
+Example:
+
+```js
+import { withIsServer } from "web/core/branchServerClient";
+
+const MyIsomorphicComponent => props => <div>
+  My latest render occured on the {props.isServer ? "server" : "client" }.
+</div>;
+
+export default withIsServer()(MyIsomorphicComponent);
+```
+
+<blockquote class="note">
+  **ProTip™:** in order to reduce the amount of Javascript sent to users, we've found that code using `branchServerClient()` are often good candidates to code splitting. You may consider using a `loadable()` component here. You can use [`DEBUG="front-commerce:webpack"`](/docs/reference/environment-variables.html#Debugging) to analyze your bundles and decide wether it would improve your application or not.
+</blockquote>
+
```

Comment:
Should we mention the fact that the server part will always be rendered first before rendering the client? This stands even if the user is navigating client side.

Reply:
What are the pitfalls of this? I am not sure how to mention it here without making the whole page too cumbersome.

I'd appreciate a hand for this, to see how it could fit in the big picture. I suggest to add it in an upcoming PR after merging this one.